site/src/pages/RoadmapPage/Course.tsx

```diff
@@ -43,7 +43,7 @@ const Course: FC<CourseProps> = (props) => {
     <div className={`course ${requiredCourses ? 'invalid' : ''}`}>
       <div className="course-card-top">
         <div className="course-and-info">
-          <div className="name">{department + ' ' + number}</div>
+          <a className="name" href={'/course/' + props.department.replace(/\s+/g, '') + props.number.replace(/\s+/g, '')} target="_blank">{department + ' ' + number}</a>
```

Comment:
I suggest wrapping the logic to create the `href` inside a helper function, kind of like how this [PR](https://github.com/icssc/peterportal-client/pull/264/files#diff-d346370d871034780e84d0cf003a1049587ebc3b242be456c9efb3779345b396R238) encapsulates the logic for `initializeYear`. 

I'll leave the choice up to you, but otherwise great work!